From 56b81cc1774ba3c56c0d04cd738d4509c1d05cac Mon Sep 17 00:00:00 2001
From: Codrin Ciubotariu <codrin.ciubotariu@microchip.com>
Date: Mon, 21 Mar 2022 18:27:07 +0200
Subject: [PATCH 368/471] ARM: dts: at91: sama7g5ek: add node for Microchip's
 ASRC card

Add and use the new sound card that enables SAMA7G5's ASRC.

Signed-off-by: Codrin Ciubotariu <codrin.ciubotariu@microchip.com>
---
 arch/arm/boot/dts/at91-sama7g5ek.dts | 26 ++++++++++++++++++++++----
 1 file changed, 22 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boot/dts/at91-sama7g5ek.dts b/arch/arm/boot/dts/at91-sama7g5ek.dts
index 4ee45c8ac4ec..13a212961e6a 100644
--- a/arch/arm/boot/dts/at91-sama7g5ek.dts
+++ b/arch/arm/boot/dts/at91-sama7g5ek.dts
@@ -14,6 +14,7 @@
 #include <dt-bindings/mfd/atmel-flexcom.h>
 #include <dt-bindings/input/input.h>
 #include <dt-bindings/pinctrl/at91.h>
+#include <dt-bindings/sound/microchip,asrc-card.h>
 #include <dt-bindings/sound/microchip,pdmc.h>
 
 / {
@@ -89,12 +90,21 @@ memory@60000000 {
 	};
 
 	sound: sound {
-		compatible = "simple-audio-card";
-		simple-audio-card,name = "sama7g5ek audio";
 		#address-cells = <1>;
 		#size-cells = <0>;
-		simple-audio-card,dai-link@0 {
+		assigned-clocks = <&pmc PMC_TYPE_CORE PMC_AUDIOPMCPLL>;
+		assigned-clock-rates = <196608000>;
+		compatible = "microchip,asrc-card";
+		microchip,model = "mchp-asrc-card @ sama5g7 EK";
+		status = "okay";
+
+		microchip,audio-asrc = <&asrc 0>, <&asrc 1>, <&asrc 2>, <&asrc 3>;
+
+		microchip,dai-link@0 {
 			reg = <0>;
+			microchip,convert-channels = <2>;
+			microchip,convert-rate = <48000>;
+			microchip,convert-sample-format = <MCHP_ASRC_PCM_FORMAT_S24_LE>;
 			cpu {
 				sound-dai = <&spdiftx>;
 			};
@@ -102,8 +112,12 @@ codec {
 				sound-dai = <&spdif_out>;
 			};
 		};
-		simple-audio-card,dai-link@1 {
+
+		microchip,dai-link@1 {
 			reg = <1>;
+			microchip,convert-channels = <2>;
+			microchip,convert-rate = <48000>;
+			microchip,convert-sample-format = <MCHP_ASRC_PCM_FORMAT_S24_LE>;
 			cpu {
 				sound-dai = <&spdifrx>;
 			};
@@ -124,6 +138,10 @@ spdif_out: spdif-out {
 	};
 };
 
+&asrc {
+	status = "okay";
+};
+
 &adc {
 	vddana-supply = <&vddout25>;
 	vref-supply = <&vddout25>;
-- 
2.34.1

